src/target/riscv/riscv.c

```diff
@@ -544,6 +544,37 @@ static int maybe_add_trigger_t2(struct target *target,
 	if (trigger->write)
 		tdata1 |= CSR_MCONTROL_STORE;
 
+	if (trigger->execute == false) {
+		/* set size for watchpoint */
+		const unsigned stab[] = {
+			0, /* 0 bits  0 bytes */
```

Comment:
Value `tdata1.size=0` does not mean "0 bits" but "match against access of any size".

Reply:
I used 0 as an illegal value because a watchpoint of size 0 should not be present. I will use -1 to identify illegal values, new patch will be submitted later